src/instruments/pfeiffer/tpg36x.py

```diff
@@ -0,0 +1,364 @@
+            if not isinstance(value, self.SensorStatus):
+                raise ValueError("The status must be a SensorStatus enum.")
+            if value == self.SensorStatus.CANNOT_TURN_ON_OFF:
+                raise ValueError("You cannot set the status to this value.")
+            status_to_send = [0 for _ in range(self._parent.number_channels)]
+            status_to_send[self._chan] = value.value
+            status_to_send_str = ",".join([str(x) for x in status_to_send])
+            self._parent.sendcmd(f"SEN,{status_to_send_str}")
+
+    @property
+    def channel(self):
+        """
+        Gets a specific channel object.
+
+        Note that the channel number is pythonic, i.e., the first channel is 0.
+
+        :rtype: `TPG36x.Channel`
+
+        Example:
+            >>> import instruments as ik
+            >>> inst = ik.pfeiffer.TPG36x.open_serial("/dev/ttyUSB0", 9600)
+            >>> ch = inst.channel[0]
+        """
+        return ProxyList(self, self.Channel, range(self._number_channels))
+
+    @property
+    def ethernet_configuration(self):
+        """
+        Get / set the ethernet configuration of the TPG36x.
+
+        .. note:: If you set the configuration to DHCP, you can simply send
+            `TPG36x.EthernetMode.DHCP` as the sole value. To set it to static,
+            you must provide a list of 4 elements: `[EthernetMode, IP, Subnet, Gateway]`.
+            The types are as follows: `TPG36x.EthernetMode`, `str`, `str`, `str`.
+
+        :return: List of the current configuration:
+            0. Configuration enum `TPG36x.EthernetMode`
+            1. IP address as string
+            2. Subnet mask as string
+            3. Gateway as string
+
+        Example:
+            >>> import instruments as ik
+            >>> inst = ik.pfeiffer.TPG36x.open_serial("/dev/ttyUSB0", 9600)
+            >>> inst.ethernet_configuration = [inst.EthernetMode.STATIC, "192.168.1.42", "255.255.255.0", "192.168.1.1"]
+           >>> inst.ethernet_configuration
+            [inst.EthernetMode.STATIC, "192.168.1.42", "255.255.255.0", "192.168.1.1"]
+        """
+        return_list = self.query("ETH").split(",")
+        return_list[0] = self.EthernetMode(int(return_list[0]))
+        return return_list
+
+    @ethernet_configuration.setter
+    def ethernet_configuration(self, value):
+        if not isinstance(value, list) or len(value) != 4:  # check for correct format
+            if value == self.EthernetMode.DHCP:  # DHCP is a special case
+                value = [self.EthernetMode.DHCP, "0.0.0.0", "0.0.0.0", "0.0.0.0"]
+            else:
+                raise ValueError(
+                    "The ethernet configuration must be a list of 4 elements."
+                )
+        if not isinstance(value[0], self.EthernetMode):  # check for correct type
+            raise ValueError("The first element must be an EthernetMode.")
+
+        for addr in value[1:]:
+            try:
+                addr = addr.split(".")
```

Comment:
consider using the `ipaddress` standard library to do this validation work https://docs.python.org/3/library/ipaddress.html

Reply:
great idea, i'll change that and leave the testing up to the standard library :)

Reply:
Done